python/cudf/cudf/core/frame.py

```diff
@@ -480,6 +480,11 @@ def _concat(
                     cudf.core.index.as_index(out.index._values)
                 )
 
+        # Reassign precision for any decimal cols
+        for name, col in out._data.items():
+            if isinstance(col, cudf.core.column.DecimalColumn):
+                col.dtype.precision = tables[0]._data[name].dtype.precision
```

Comment:
Does this mean that we always just take the precision from the 0'th object in the list as the resulting precision, and if so, is that safe?

Reply:
Yep, because the call to `_cast_cols_to_common_dtypes` both ensures that all of the decimal columns to be concatenated have the same dtype and that any missing columns are filled with all-null columns of the same dtype